src/librustc/hir/mod.rs

```diff
@@ -388,24 +388,33 @@ impl PathSegment {
     }
 }
 
+#[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
+pub struct ConstArg {
+    pub value: AnonConst,
+    pub span: Span,
```

Comment:
The expression span is available through `AnonConst`, is it necessary to duplicate it here?
(If not, then `struct ConstArg` can be refactored away as well.)

Reply:
Ah, I see, it's available, but not without side tables.